eo-maven-plugin/src/main/java/org/eolang/maven/ParseMojo.java

```diff
@@ -104,85 +98,81 @@ public final class ParseMojo extends SafeMojo {
-                        }
-                    }
-                    tasks.add(
-                        Executors.callable(
-                            () -> {
-                                try {
-                                    this.parse(tojo);
-                                    total.incrementAndGet();
-                                } catch (final IOException ex) {
-                                    throw new IllegalStateException(
-                                        String.format(
-                                            "Unable to parse %s",
-                                            tojo.get(Tojos.KEY)
-                                        ),
-                                        ex
-                                    );
-                                }
-                            }
-                        )
-                    );
-                }
-            );
-        try {
-            for (final Future<Object> completed : Executors.newFixedThreadPool(this.threads)
-                .invokeAll(tasks)) {
-                try {
-                    completed.get();
-                } catch (final ExecutionException ex) {
-                    throw new IllegalArgumentException(
-                        ex.getCause().getMessage(),
-                        ex
-                    );
-                }
-            }
-        } catch (final InterruptedException ex) {
-            Thread.currentThread().interrupt();
-            throw new IllegalStateException(
-                String.format(
-                    "Interrupted while waiting for %d parsing to finish",
-                    total.get()
-                ),
-                ex
-            );
-        }
-        if (total.get() == 0) {
-            if (tojos.isEmpty()) {
+        final int total = tasks
+            .parallelStream()
+            .mapToInt(Supplier::get)
+            .sum();
+        if (0 == total) {
+            if (((Collection<Tojo>) this.scopedTojos().select(
+                row -> row.exists(AssembleMojo.ATTR_EO)
+            )).isEmpty()) {
                 Logger.info(this, "No .eo sources need to be parsed to XMIRs");
             } else {
                 Logger.info(this, "No .eo sources parsed to XMIRs");
             }
         } else {
-            Logger.info(this, "Parsed %d .eo sources to XMIRs", total.get());
+            Logger.info(this, "Parsed %d .eo sources to XMIRs", total);
+        }
+    }
+
+    /**
+     * Create a task for parsing a single EO file.
+     *
+     * @param tojo Tojo
+     * @return Task
+     */
+    private Supplier<Integer> task(final Tojo tojo) {
+        final ClassLoader loader = Thread.currentThread().getContextClassLoader();
+        return () -> {
+            try {
+                Thread.currentThread().setContextClassLoader(loader);
+                this.parse(tojo);
+                return 1;
+            } catch (final IOException ex) {
```

Comment:
@volodya-lombrozo in general, it's a bad practice to use `catch` inside mid-level methods. See https://www.yegor256.com/2015/07/09/catch-if-cant-otherwise.html Every time you do this, ask yourself: maybe there is a better way? In this case, don't use `Supplier`, use `Scalar` from Cactoos. Or, even better, instead of `parallelStreams()` use `Threads` from Cactoos -- they do the same, but in more OOP way and more transparently, I believe.

Reply:
@yegor256 done